api4/system.go

```diff
@@ -515,3 +517,38 @@ func getRedirectLocation(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(model.MapToJson(m)))
 	return
 }
+
+func pushNotificationsAck(c *Context, w http.ResponseWriter, r *http.Request) {
+	ack := model.PushNotificationAckFromJson(r.Body)
+
+	if !*c.App.Config().EmailSettings.PushNotificationsAck {
+		c.Err = model.NewAppError("pushNotificationAck", "api.push_notifications_ack.disabled.app_error", nil, "", http.StatusForbidden)
+		return
+	}
+
+	var err *model.AppError
+	var notification *model.PushNotification
+	ackAt := model.GetMillis()
+
+	if pluginsEnvironment := c.App.GetPluginsEnvironment(); pluginsEnvironment != nil {
+		pluginContext := c.App.PluginContext()
+		pluginsEnvironment.RunMultiPluginHook(func(hooks plugin.Hooks) bool {
+			notification, err = hooks.PushNotificationAck(pluginContext, ack.Id, ack.DeviceId, ack.ReceivedAt, ackAt)
```

Comment:
Why do you need the device ID? the ack.Id should be enough

Reply:
The `notificationId` is shared by all the devices, so... if you want to know the exact received notification you need both, the deviceId and the notificationId. We could add another generated id if it is necessary for each device notification.

Reply:
Using a unique ack Id for each push notification (not sharing it between devices).